### PR TITLE
Recommend host hardening policies

### PR DESCRIPTION
**Pull request by vishnusomank**
- show KubearmorHostPolicies while recommending hardening policies based on flag value

Signed-off-by: Vishnu Soman <vishnu@accuknox.com>

### CODE REVIEW

File: src/config/configManager.go [hunk 1]
```diff
@@ -512,3 +513,7 @@ func GetCfgRecCronJobTime() string {
 func GetCfgRecOneTime() string {
 	return CurrentCfg.ConfigRecommendPolicy.OneTimeJobTimeSelection
 }
+
+func GetRecommendHost() bool {
```
Comment (seswarrajan) [inline src/config/configManager.go]:
```suggestion
func GetCfgRecommendHostPolicy() bool {
```

File: src/recommendpolicy/helperFunctions.go [hunk 2]
```diff
@@ -73,6 +74,21 @@ func generatePolicy(name, namespace string, labels LabelMap) ([]types.KnoxSystem
 				return []types.KnoxSystemPolicy{}, err
 			}
 			policies = append(policies, policy)
+		} else if ms.Kind == types.KindKubeArmorHostPolicy && cfg.GetRecommendHost() {
+			client := cluster.ConnectK8sClient()
+			nodeList, err := client.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
```
Comment (seswarrajan) [inline src/recommendpolicy/helperFunctions.go]: Ant client go APIs/watcher/listeners, please take in under customResourceHandler.go and call the API from there.

**Review by seswarrajan — CHANGES_REQUESTED** [review]

**Review by seswarrajan — APPROVED** [review]
LGTM

**Comment by seswarrajan** [comment]
@vishnusomank Please fix gosec issues

**Review by seswarrajan — APPROVED** [review]
LGTM.
Is this validated with the latest change ?

Since it is host policy, is it tested with running as systemd in local machine ?

**Comment by vishnusomank** [comment]
> LGTM. Is this validated with the latest change ?
> 
> Since it is host policy, is it tested with running as systemd in local machine ?

yes this was tested locally on a k3s cluster @seswarrajan

**Review by achrefbensaad — APPROVED** [review]
Can you please fix lint error. else LGTM!

For some reason I can't add code comments.
https://github.com/accuknox/discovery-engine/pull/662/files#diff-e482c14daf9bbc77d728fa6a131912cb553e5652f4f31572b6c6da87ae148d4cR1

File: src/logging/logger.go [hunk 3]
```diff
@@ -53,7 +54,7 @@ func configure(config Config) *zerolog.Logger {
 }
 
 func newRollingFile(config Config) io.Writer {
-	if err := os.MkdirAll(config.Directory, 0744); err != nil {
+	if err := os.MkdirAll(config.Directory, 0740); err != nil {
```
Comment (seswarrajan) [inline src/logging/logger.go:57]: @vishnusomank Why do re provide execute permission for the user ?
It could be 640 right ?

Comment (vishnusomank) [reply src/logging/logger.go:57]: yes 640 would be enough. I changed it into 740 from 744 to pass lint. I can change it into 640 if required @seswarrajan

Comment (vishnusomank) [reply src/logging/logger.go:57]: @seswarrajan i've tested it by changing permission into 0640. But some of the functionalities were failing with error `permission denied`. can we keep it 740 itself?

Comment (seswarrajan) [reply src/logging/logger.go:57]: 740 is fine

**Comment by seswarrajan** [comment]
Kindly resolve conflicts and update.

**Comment by vishnusomank** [comment]
> Kindly resolve conflicts and update.

@seswarrajan i've resolved the conflicts and updated. kindly review

**Review by seswarrajan — APPROVED** [review]
LGTM

**Pull request merged**